src/test/unit_tests/test_bucketspace_versioning.js

```diff
@@ -2635,6 +2774,36 @@ mocha.describe('bucketspace namespace_fs - versioning', function() {
                 assert.equal(err.name, 'NotFound');
             }
         });
+
+        mocha.it('head object - versioning enabled - nested key (more than 1 level)', async function() {
+            const body1 = 'A'.repeat(5);
+            const body2 = body1 + 'B'.repeat(3);
+            const body3 = body2 + 'C'.repeat(7);
+            const dir_path_nested = 'documents/finance/2024/October/';
+            const nested_key_level4 = path.join(dir_path_nested, 'report.pdf');
+            await s3_client.putObject({ Bucket: nested_keys_bucket_name, Key: nested_key_level4, Body: body1 });
+            // only after the second PUT object we expect to have the .versions under the parent directory of the file
+            const put_res2 = await s3_client.putObject({ Bucket: nested_keys_bucket_name, Key: nested_key_level4, Body: body2});
+            await s3_client.putObject({ Bucket: nested_keys_bucket_name, Key: nested_key_level4, Body: body3});
+
+            // latest
+            const head_res = await s3_client.headObject({Bucket: nested_keys_bucket_name, Key: nested_key_level4});
+            assert.equal(head_res.ContentLength, body3.length);
+
+            // by version-id
+            const head_res2 = await s3_client.headObject({
+                Bucket: nested_keys_bucket_name, Key: nested_key_level4, VersionId: put_res2.VersionId});
+            assert.equal(head_res2.ContentLength, body2.length);
+
+            // partial path of (directory without the key)
+            try {
+                await s3_client.headObject({Bucket: nested_keys_bucket_name, Key: dir_path_nested});
```

Comment:
I suggest start using `assert.rejects`, I think it would cleanup our code

Reply:
It would be: `await assert.rejects(async () => await s3_client.headObject({Bucket: nested_keys_bucket_name, Key: dir_path_nested}));` instead of the try..catch clause, although it is one-liner, it is less flexible to debugging in my opinion.

Reply:
per our discussion on slack, I don't think there are missing debugging features, here are some examples for debugging - https://www.geeksforgeeks.org/node-js-assert-rejects-function/
but it's a matter of styling

Reply:
At this point I didn't change the code, maybe in the future I will try it again.